### PR TITLE
Limit lists to 10,000 items in fastbinary decoding to avoid crashing …

### PR DESCRIPTION
**Pull request by dvirsky**
…servers from huge allocations on junk/mallicious input

https://issues.apache.org/jira/browse/THRIFT-3175

### CODE REVIEW

**Comment by dvirsky** [comment]
The build failure seems to be unrelated to the patch:

```
...
vector-0.10.12.3 failed during the building phase. The exception was:
ExitFailure 9
This may be due to an out-of-memory condition.
```

File: lib/py/src/protocol/fastbinary.c [hunk 1]
```diff
@@ -936,7 +951,7 @@ decode_struct(DecodeBuffer* input, PyObject* output, PyObject* spec_seq) {
 }
 
 
-/* --- MAIN RECURSIVE INPUT FUNCTION --- */
+/* --- MAIN RECURSIVE INPUT FUCNTION --- */
```
Comment (bufferoverflow) [inline lib/py/src/protocol/fastbinary.c]: What's this?

File: lib/py/src/protocol/fastbinary.c [hunk 2]
```diff
@@ -1028,7 +1043,7 @@ decode_val(DecodeBuffer* input, TType type, PyObject* typeargs) {
     }
 
     len = readI32(input);
-    if (!check_ssize_t_32(len)) {
+    if (!check_list_length(len)) {
       return NULL;
```
Comment (bufferoverflow) [inline lib/py/src/protocol/fastbinary.c]: I guess we need to inform the user about reaching the limit somehow, e.g. via PyErr_SetString

File: lib/py/src/protocol/fastbinary.c (hunk 1 above)
Comment (dvirsky) [inline lib/py/src/protocol/fastbinary.c]: I'll fix the typo

File: lib/py/src/protocol/fastbinary.c (hunk 2 above)
Comment (dvirsky) [inline lib/py/src/protocol/fastbinary.c]: check_list_length already does that:
```c
   PyErr_SetString(PyExc_OverflowError, "list size out of the sanity limit (10000 items max)");
```

File: lib/py/src/protocol/fastbinary.c (hunk 1 above)
Comment (dvirsky) [inline lib/py/src/protocol/fastbinary.c]: fixed in 9994c6e55f3c11acc523545c1ca67deb05cd0910

**Pull request closed without merging**